wordpress-mobile/WordPressAuthenticator-iOS
# Improves our suggestions for bad URLs in LoginSiteAddressViewController.

## Description:

This PR improves our suggestions for bad URLs in `LoginSiteAddressViewController`, as discussed here: https://github.com/wordpress-mobile/WordPress-iOS/issues/10294

We no longer allow the user to tap "Next" unless they input a valid URL into the field (check "expectations" below for exceptions).

We also show an error message for invalid URLs after 2 seconds if there's no activity, or as soon as the user taps outside of the URL field as shown in the demo below.

## Demo

![56679116-b178ee00-669a-11e9-975e-2781a5130797](https://user-images.githubusercontent.com/1836005/56696037-1e9f7a00-66c1-11e9-8ebf-28a54429efd7.gif)

## Expectations

URL validation is complex.  This PR doesn't expect to catch all invalid URLs as invalid, but at least the most obvious ones.

These changes however should not prevent you from inserting ANY valid URL into the field and clicking next.

## Testing:

You can test this PR through WPiOS's branch `issue/10294-better-error-for-invalid-url`.

1. Run WPiOS
2. At the login screen select "Login with Site Address".
3. Start playing with the URL field and make sure it works as described in this PR, and a shown in the demo above.
